### PR TITLE
Switch to maven resolver from aether

### LINKED ISSUE #245: Upgrade to use https://maven.apache.org/resolver/ instead of old aether-impl
Maven (the build tool) does not directly import the aether-impl jar.  It has created a new project (with the same aether classes) here - https://github.com/apache/maven-resolver

This should be considered the 'defacto standard' for maven artifact resolution.

### PR DESCRIPTION
**Pull request by markpollack**
Fixes #245

### CODE REVIEW

**Comment by markpollack** [comment]
Should use Logger instead of direct print

File: spring-cloud-deployer-resource-maven/src/main/java/org/springframework/cloud/deployer/resource/maven/ConsoleRepositoryListener.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+	public void metadataResolving( RepositoryEvent event )
+	{
+		println( "metadataResolving", event.getMetadata() + " from " + event.getRepository() );
+	}
+
+	private void println( String event, String message )
+	{
+		out.println( "Aether Repository - " + event + ": " + message );
+	}
+}
```
Comment (dturanski) [inline spring-cloud-deployer-resource-maven/src/main/java/org/springframework/cloud/deployer/resource/maven/ConsoleRepositoryListener.java]: Add new line

Comment (dturanski) [reply spring-cloud-deployer-resource-maven/src/main/java/org/springframework/cloud/deployer/resource/maven/ConsoleRepositoryListener.java]: Agree with using Logger

**Comment by cppwfs** [comment]
With the replacement of aether with maven, the link to https://github.com/eclipse/aether-core/blob/4cf5f7a406b516a45d8bf15e7dfe3fb3849cb87b/aether-api/src/main/java/org/eclipse/aether/repository/RepositoryPolicy.java#L16 in the configuration.adoc from the  SCDF reference documentation needs to be updated.

**Comment by ilayaperumalg** [comment]
LGTM, merging.

**Comment by ilayaperumalg** [comment]
Squashed and merged as de4930e

**Pull request closed without merging**